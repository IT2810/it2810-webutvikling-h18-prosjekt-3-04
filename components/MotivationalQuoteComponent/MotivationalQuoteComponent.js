import React from 'react';
import styles from "./styles/styles";
import {Text, View} from "react-native";
import PropTypes from 'prop-types'

export default class MotivationalQuoteComponent extends React.Component {

    constructor(props) {
        super(props);
        this.newQuote = this.newQuote.bind(this);
        this.state = {
            quote: 'No quote given',
        };
    }

    componentDidMount() {
        this.newQuote();
    }

    static propTypes = {
        numTodos: PropTypes.number,
    };

    motivationalQuotes = [
        "Just do it!",
        "You are doing great!",
        "You are great!",
        "You go girl!",
        "Stay productive!",
        "Go get ‘em!",
        "Nice progression!",
        "Achieve your goals!",
        "You become great.",
        "Move along.",
        "How are you?",
        "Andreas loves you!",
        "Gotta catch 'em all!",
        "Go go gadget!",
        "Nothing is impossible.",
        "Live in the present.",
        "Apples are nice.",
        "I like you.",
        "Do you like me?",
        "Run, Forrest!",
        "Catch me if you can.",
        "What are your goals?",
        "Do you remember?",
        "Have you done it yet?",
        "Stop procrastinating!",
        "Stay hydrated!",
        "B U T L E R ❤️ you."
    ];

    newQuote = () => {
        if (this.props.numTodos > 0) {
            let rng = Math.floor(Math.random() * (this.motivationalQuotes.length - 1));
            this.setState({ quote: this.motivationalQuotes[rng] });
        } else {
            this.setState({ quote: 'Looks like you have no tasks. Press the button in the lower right corner to get started :)' });
        }

    };

    render() {
        return(
            <View style={styles.motivationalElement}>
                <Text style={styles.motivationalText}>{this.state.quote}</Text>
            </View>
        );
    }
}