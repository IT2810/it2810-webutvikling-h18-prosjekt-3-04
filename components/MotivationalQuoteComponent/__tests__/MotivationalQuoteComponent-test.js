import React from 'react';
import MotivationalQuoteComponent from '../MotivationalQuoteComponent';
import renderer from 'react-test-renderer';


test('renders correctly', () => {
    const testCreateTaskViewComponent = <MotivationalQuoteComponent/>;
    const tree = renderer.create(testCreateTaskViewComponent).toJSON();
    expect(tree).toMatchSnapshot();
});
