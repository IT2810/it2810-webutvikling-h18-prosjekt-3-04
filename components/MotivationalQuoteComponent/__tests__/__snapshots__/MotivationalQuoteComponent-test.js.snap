// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "height": 90,
      "marginTop": 15,
      "width": 710,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "alignItems": "flex-start",
        "color": "#bcbcbc",
        "flex": 1,
        "fontSize": 16,
        "justifyContent": "center",
        "marginLeft": 15,
        "marginTop": 32,
      }
    }
  />
</View>
`;
